std-bits/table/src/main/java/org/enso/table/write/ExcelWriter.java

```diff
@@ -56,61 +55,61 @@ public static void writeTableToSheet(Workbook workbook, int sheetIndex, boolean
       }
 
       writeTableToSheet(workbook, sheet, firstRow, 1, table, rowLimit, headers != ExcelHeaders.HeaderBehavior.EXCEL_COLUMN_NAMES);
-    } else if (replace) {
-      if (headers == ExcelHeaders.HeaderBehavior.INFER) {
-        ExcelSheet excelSheet = new ExcelSheet(workbook, sheetIndex);
-        headers = shouldWriteHeaders(excelSheet, firstRow + 1, 1, -1)
-            ? ExcelHeaders.HeaderBehavior.USE_FIRST_ROW_AS_HEADERS
-            : ExcelHeaders.HeaderBehavior.EXCEL_COLUMN_NAMES;
-      }
+    } else if (existingDataMode == ExistingDataMode.REPLACE) {
+      headers = headers != ExcelHeaders.HeaderBehavior.INFER ? headers :
+          shouldWriteHeaders(new ExcelSheet(workbook, sheetIndex), firstRow + 1, 1, -1);
 
       String sheetName = workbook.getSheetName(sheetIndex - 1);
       workbook.removeSheetAt(sheetIndex - 1);
 
       Sheet sheet = workbook.createSheet(sheetName);
       workbook.setSheetOrder(sheetName, sheetIndex - 1);
       writeTableToSheet(workbook, sheet, firstRow, 1, table, rowLimit, headers != ExcelHeaders.HeaderBehavior.EXCEL_COLUMN_NAMES);
-    } else {
+    } else if (existingDataMode == ExistingDataMode.ERROR){
       throw new ExistingDataException("Sheet already exists, and cannot be replaced in current mode.");
+    } else {
+      // In Append Mode, so lets go to a Range based approach.
```

Comment:
This comment is not very clear to me. Can you maybe elaborate what range-based approach means there?